tests/valid_cases/input.rs

```diff
@@ -198,12 +209,154 @@ fn contract() {
         )
         .err()
         .unwrap();
+
     assert_eq!(
         ValidationError::InputContractAssociatedOutputContract { index: 1 },
         err
     );
 }
 
+#[test]
+fn message() {
+    let rng = &mut StdRng::seed_from_u64(8586);
+
+    let txhash: Bytes32 = rng.gen();
+
+    let predicate = generate_nonempty_padded_bytes(rng);
+    let owner = (*Contract::root_from_code(&predicate)).into();
+
+    Input::message_predicate(
+        rng.gen(),
+        rng.gen(),
+        rng.gen(),
+        rng.gen(),
+        rng.gen(),
+        owner,
+        generate_bytes(rng),
+        predicate,
+        generate_bytes(rng),
+    )
+    .validate(1, &txhash, &[], &[], &Default::default())
+    .expect("failed to validate empty message input");
+
+    let mut tx = Transaction::default();
+
+    let input = Input::message_signed(
+        rng.gen(),
+        rng.gen(),
+        rng.gen(),
+        rng.gen(),
+        rng.gen(),
+        rng.gen(),
+        0,
+        generate_bytes(rng),
+    );
+
+    tx.add_input(input);
+
+    let block_height = rng.gen();
+    let err = tx
+        .validate(block_height, &Default::default())
+        .err()
+        .expect("Expected failure");
```

Comment:
Needs docs clarifying what the expected failure actually is here

```suggestion
        .expect("Expected failure");
```